src/test/java/org/reaktivity/specification/socks/internal/SocksFunctionsTest.java

```diff
@@ -44,43 +48,136 @@ public void setUp() throws Exception
+
+        assertNotNull(routeExBytes);
+    }
+
+    @Test
+    public void shouldBuildRouteExWithIpv4Address() throws Exception
+    {
+        byte[] bytes = SocksFunctions.routeEx()
+                                     .address("127.0.0.1")
+                                     .port(8080)
+                                     .build();
+        DirectBuffer buffer = new UnsafeBuffer(bytes);
+        SocksRouteExFW routeEx = new SocksRouteExFW().wrap(buffer, 0, buffer.capacity());
+
+        assertEquals(KIND_IPV4_ADDRESS, routeEx.address().kind());
+        assertArrayEquals(new byte[] { 127, 0, 0, 1 },
+            copyOfRange(routeEx.buffer().byteArray(), routeEx.address().offset() + 1, routeEx.address().limit()));
+        assertEquals(8080, routeEx.port());
+    }
+
+    @Test
+    public void shouldBuildBeginExWithIpv4Address() throws Exception
+    {
+        byte[] bytes = SocksFunctions.beginEx()
+                                     .typeId(0)
+                                     .address("127.0.0.1")
+                                     .port(8080)
+                                     .build();
+        DirectBuffer buffer = new UnsafeBuffer(bytes);
+        SocksBeginExFW beginEx = new SocksBeginExFW().wrap(buffer, 0, buffer.capacity());
+
+        assertEquals(KIND_IPV4_ADDRESS, beginEx.address().kind());
+        assertArrayEquals(new byte[] { 127, 0, 0, 1 },
+            copyOfRange(beginEx.buffer().byteArray(), beginEx.address().offset() + 1, beginEx.address().limit()));
+        assertEquals(8080, beginEx.port());
+    }
+
+    @Test
+    public void shouldLoadFunctionsWithIpv6Address() throws Exception
+    {
+        String expressionText = "${socks:routeEx()" +
+                                       ".address(\"2001:0db8:85a3:0000:0000:8a2e:0370:7334\")" +
+                                       ".port(8080)" +
+                                       ".build()}";
+        ValueExpression expression = factory.createValueExpression(ctx, expressionText, String.class);
+        String routeExBytes = (String) expression.getValue(ctx);
+
         assertNotNull(routeExBytes);
     }
 
     @Test
-    public void shouldBuildRouteEx() throws Exception
+    public void shouldBuildRouteExWithIpv6Address() throws Exception
     {
         byte[] bytes = SocksFunctions.routeEx()
-                                     .address("example.com")
+                                     .address("2001:0db8:85a3:0000:0000:8a2e:0370:7334")
                                      .port(8080)
                                      .build();
         DirectBuffer buffer = new UnsafeBuffer(bytes);
         SocksRouteExFW routeEx = new SocksRouteExFW().wrap(buffer, 0, buffer.capacity());
 
-        assertEquals("example.com", routeEx.address().asString());
+        assertEquals(KIND_IPV6_ADDRESS, routeEx.address().kind());
+        assertArrayEquals(new byte[] { 32, 1, 13, -72, -123, -93, 0, 0, 0, 0, -118, 46, 3, 112, 115, 52 },
```

Comment:
Given that IPv6 addresses are defined in hexadecimal, recommend replacing `new byte[] { ... }` with `new BigInteger("20010db885a3000000008a2e03707334", 16).toByteArray()`

Please apply same feedback to other similar tests.

Reply:
done